File: Tests/RegexAdapterTests.cs
using System;
using System.Text.Patterns;
using System.Text.RegularExpressions;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Tests {
	[TestClass]
	public class RegexAdapterTests {
		private Pattern pattern;

		[TestMethod]
		public void Converter() {
			Assert.ThrowsException<PatternConstructionException>(() => pattern = new Regex(@"hello"));
			pattern = new Regex(@"^hello");
			Assert.ThrowsException<PatternConstructionException>(() => pattern = new Regex(@"hello$"));
			pattern = new Regex(@"^hello$");
		}

		[TestMethod]
		public void Consume() {
			Pattern Regex = new Regex(@"^hello");
			ResultAssert.Captures("hello", Regex.Consume("hello"));
			ResultAssert.Captures("hello", Regex.Consume("hello world"));
			ResultAssert.Fails(Regex.Consume("hel"));
		}
	}
}

File: Patterns.Tests/RegexAdapterTests.fs
namespace Patterns

open System
open System.Text.RegularExpressions
open Stringier.Patterns
open Microsoft.VisualStudio.TestTools.UnitTesting

[<TestClass>]
type RegexAdapterTests() =

    [<TestMethod>]
    member _.``converter`` () =
        let mutable pattern = (fun () -> Regex(@"hello").AsPattern() |> ignore)
        Assert.ThrowsException<PatternConstructionException>(Action pattern) |> ignore
        Regex(@"^hello") |> ignore
        pattern <- (fun () -> Regex(@"hello$").AsPattern() |> ignore)
        Assert.ThrowsException<PatternConstructionException>(Action pattern) |> ignore
        Regex(@"^hello$") |> ignore
        
    [<TestMethod>]
    member _.``consume`` () =
        let pattern = Regex(@"^hello").AsPattern()
        ResultAssert.Captures("hello", pattern.Consume("hello"))
        ResultAssert.Captures("hello", pattern.Consume("hello world"))
        ResultAssert.Fails(pattern.Consume("hel"))